scripts/readelf.py

```diff
@@ -251,20 +252,24 @@ def display_program_headers(self, show_heading=True):
             (Elf file type is...)
         """
         self._emitline()
-        if self.elffile.num_segments() == 0:
+        nseg = self.elffile.num_segments()
```

Comment:
I'd rather you just inline the call everywhere, this isn't much of a performance optimization anyhow